resource/global/YoStarEN/resource/tasks.json

```diff
@@ -7031,6 +7031,321 @@
             72
         ]
     },
+    "Sami@Roguelike@CheckCollapsalParadigms_onStage": {
+        "algorithm": "OcrDetect",
+        "action": "DoNothing",
+        "cache": false,
+        "text": [
+            "初霜湖泽",
+            "密静林地",
+            "昧明冻土",
+            "积冰岩骸",
+            "无瑕花园",
+            "远见之构",
+            "永恒之尘",
+            "深埋迷境"
+        ],
+        "ocrReplace": [
+            [
+                "Brisk Marshalands",
+                "初霜湖泽" 
+            ],
+            [
+                "Placid Forests",
+                "密静林地" 
+            ],
+            [
+                "Dawning Frostbounds",
+                "昧明冻土" 
+            ],
+            [
+                "Gelid Rockshelves",
+                "积冰岩骸" 
+            ],
+            [
+                "Immaculate Garden",
+                "无瑕花园" 
+            ],
+            [
+                "Clayrovant Carcass",
+                "远见之构" 
+            ],
+            [
+                "Secluded Passage",
+                "深埋迷境" 
+            ]
```

Comment:
Unfortunately, this will most likely not work.
In IS4 (Yostar only knows why) the text is not displayed with a new line. It's all a single line, making the text much smaller and almost too hard for the OCR to detect. 
In some cases I had to revert to use the temperature, which is also not always consistent:
![image](https://github.com/MaaAssistantArknights/MaaAssistantArknights/assets/56174894/b668c303-74c3-4eca-8829-7938cbdd17f6)
As of today I still don't have a solution. Help is greatly accepted.

Reply:
The resolution of my screenshot is 1280 * 720, not sure if it will still trigger that issue.
Is it better if I do that in 1920 * 1280?

 
<img width="646" alt="image" src="https://github.com/MaaAssistantArknights/MaaAssistantArknights/assets/23149848/8769b03f-375a-4bad-a376-e7c9b38f469b">

Reply:
Yes, 1920x1080 seems the best compromise. I also suggest using MAA with full hd. It's in the FAQ

Reply:
Thanks for all your input! And thanks a lot for the screenshots!
Currently I use these screenshots just for calculating roi so as long as the resolution is in 16:9 it will be fine.

@Constrat
Thanks for advices!
I have two questions:
1. Do you mean, if OCR failed frequently, increase the resolution of the emulator could help? That's awesome.
2. When you use temperature to determine the zone, do you use templates or just text OCR?

Reply:
Currently it's all OCR (no templates), I've never tried fully switching to temperature only as it wasn't easy. So the current solution is just a mix of the two. In case the normal "floor" name doesn't work it'll try to OCR the numbers of the temperature.

Unfortunately just increasing the resolution of the emulator is not the best solution as:
1. It highly increases cost of screencap and reduces smoothness/ frame rate
2. In YostarEN che DPI are managed awfully: this is a small example, but almost any text can suffer from this

Reply:
@Constrat I have finished refactoring my code.
It turns out switching to templates of temperature works well.
(I prepared the templates myself so it takes some time.)
I also altered the Matcher a bit so that the name of the matched template will be returned in the result.

In the long future you may want to consider combine the features of OCRer and Matcher so that we can even do text replacement on the resulted template name.